### PR TITLE
Enable Main Thread Checker by default

### PR DESCRIPTION
**Pull request by myihsan**
### Short description 📝

When we create a new project by Xcode, the Main Thread Checker is enabled by default.
So I think it may be a good idea to have the same behavior.
If there is any reason not to enable it by default, please tell me.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
  Four of the acceptance tests failed, but it seems like the problem of my local environment since the failure have no relationship with the changes I made.
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

**Review by kwridan — COMMENTED** [review]
Hi @myihsan - thanks for raising this PR!

+1 I think it makes sense to make the default schemes include the main thread diagnostics option.

I'd suggest perhaps the changes only be made to:

- `AutogeneratedSchemesProjectMapper`
- `AutogeneratedProjectSchemeWorkspaceMapper`

As that is where the default schemes are made - we could leave initialiser parameters as they were where they default to no options. Those are typically used when a scheme is manually defined in which case it may be clearer and more consistent with how the other options work. It will also ensure anyone defining their schemes manually today don't get a sudden change in behaviour they didn't expect.

**Comment by myihsan** [comment]
Hi @kwridan, thanks for your review!

> As that is where the default schemes are made - we could leave initialiser parameters as they were where they default to no options. Those are typically used when a scheme is manually defined in which case it may be clearer and more consistent with how the other options work.

I can see your point, but there are some default values that are not consistent to me. For example `revealArchiveInOrganizer`, I expect it to be false as other parameters.
So it seems fine to me to also change the default value of initializers. What do you think?

> It will also ensure anyone defining their schemes manually today don't get a sudden change in behaviour they didn't expect.

In my opinion, the users who are using the default value, are expecting the default behavior of Xcode or just don't care about that parameter. So it may not be a problem.
However, I don't have an OSS project with over 1k users. This may be just my bias. 💦

**Review by fortmarek — APPROVED** [review]
Thanks for your first contribution to the project @myihsan!

I'd probably expect to have a default Xcode value even if I define scheme manually, too.

> It will also ensure anyone defining their schemes manually today don't get a sudden change in behaviour they didn't expect.

This is true but I think it's reasonable that if user does not specify something explicitly, it should be fine if we change the default.

**Review by kwridan — APPROVED** [review]
> In my opinion, the users who are using the default value, are expecting the default behavior of Xcode or just don't care about that parameter. So it may not be a problem.

> This is true but I think it's reasonable that if user does not specify something explicitly, it should be fine if we change the default.

Good points, we're not overriding user defined preferences here.

This is good to land! thanks again @myihsan

**Comment by kwridan** [comment]
@all-contributors add @myihsan for code

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 
 ## Next
 
+- Enable Main Thread Checker by default [#2549](https://github.com/tuist/tuist/pull/2549) by [@myihsan](https://github.com/myihsan)
```
Comment (laxmorek) [inline CHANGELOG.md:7]: Just quick comment. Should it be under "Added" or "Changed"?

Comment (myihsan) [reply CHANGELOG.md:7]: Oh, sorry for the missing title.
Let me see if there is any other PR I can create and add the title in the meantime.

**Pull request merged**